### PR TITLE
Fix non-atomic volatile update

### PR DESCRIPTION
**Pull request by yjshen**

### CODE REVIEW

**Comment by merlimat** [comment]
What is the motivation for this change? There were different specific reasons for not using volatile in these components.

Volatile has a performance penalty and should only be used if strictly needed. Also, it's not a panacea that fixes all concurrency issue.

As it is, I think this commit should be reverted.

@yjshen @sijie

File: pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/GrowablePriorityLongPairQueue.java [hunk 1]
```diff
@@ -204,7 +207,7 @@ private LongPair removeAtWithoutLock(int index) {
             LongPair item = new LongPair(data[index], data[index + 1]);
             data[index] = EmptyItem;
             data[index + 1] = EmptyItem;
-            --this.size;
```
Comment (merlimat) [inline pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/GrowablePriorityLongPairQueue.java:207]: This is already protected by a write lock. No need to use updater. The volatile is only used to ensure readers are reading the last value of the size.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 2]
```diff
@@ -1759,7 +1762,7 @@ public boolean isWritable() {
     }
 
     public void startSendOperation(Producer producer, int msgSize) {
-        messagePublishBufferSize += msgSize;
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:1762]: All operations on a connection are happening on same thread

**Comment by merlimat** [comment]
I take back the argument on performance impact.. since this is only adding the static updaters. Though in most cases these updaters are not necessary because the variables are designed to be updated from a single thread or already within the context of a mutex.

**Comment by sijie** [comment]
Yeah, this change is more about adding the static updaters. It doesn't change the existing behaviors. The change here is more about making the volatile updates atomic and ensure a consistent pattern across the code. Since it is always a good practice to follow across the whole pulsar code base. Did you see any other problems with this change?

**Pull request merged**